auto3dseg/algorithm_templates/segresnet2d/scripts/algo.py

```diff
@@ -29,6 +29,20 @@ def get_gpu_available_memory():
 
 
 class Segresnet2dAlgo(BundleAlgo):
+    def pre_check_skip_algo(self):
+        if self.data_stats_files is None:
+            return
+        data_stats = ConfigParser(globals=False)
+        if os.path.exists(str(self.data_stats_files)):
+            data_stats.read_config(str(self.data_stats_files))
+        else:
+            data_stats.update(self.data_stats_files)
+        spacing = data_stats["stats_summary#image_stats#spacing#median"]
+        if len(spacing) > 2:
+            if spacing[-1] < 3 * (spacing[0] + spacing[1]) / 2:
+                self.skip_bundlegen = True
+                self.skip_info = f'2D network is skipped due to median spacing of {spacing}.'
```

Comment:
may be return True/False, instead of setting Class member variables.  Because it seems no one else is setting those vars (and otherwise they are just not set).

are you sure you're not missing some files in this PR?

Reply:
yes that's a way to do it. The only concern is that for other algorithms this pre_check_skip is not written, so I think maybe in initialization this self.skip_bundle=False will be better.

Reply:
for other algorithms you have it's implementation in the Base class to return False, I think it's enough. 
I can also check "hasattr"  if that function exists

Reply:
@myron, thanks, I changed the function to return tuple: bool, str